com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/DotsDeformation.hlsl

```diff
@@ -0,0 +1,67 @@
+#ifndef SKINNING_INCLUDED
+#define SKINNING_INCLUDED
+
+#if defined(DOTS_INSTANCING_ON)
+struct DeformedVertexData
+{
+    float3 Position;
+    float3 Normal;
+    float3 Tangent;
+};
+
+uniform StructuredBuffer<DeformedVertexData> _DeformedMeshData;
+uniform StructuredBuffer<DeformedVertexData> _PreviousFrameDeformedMeshData;
+
+void DOTS_Deformation(inout AttributesMesh input)
+{
+    // x = curr frame index
+    // y = prev frame index
+    // z = deformation check (0 = no deformation, 1 = has deformation)
+    // w = skinned motion vectors
+    const int4 deformProperty = asint(unity_ComputeMeshIndex);
+    const int doSkinning = deformProperty.z;
+    if (doSkinning == 1)
+    {
+        const int streamIndex = _HybridDeformedVertexStreamIndex;
+        const int startIndex = deformProperty[streamIndex];
+        const DeformedVertexData vertexData = _DeformedMeshData[startIndex + input.vertexID]; //if vertexid defined
+
+        input.positionOS = vertexData.Position;
+#ifdef ATTRIBUTES_NEED_NORMAL
+        input.normalOS = vertexData.Normal;
+#endif
+#ifdef ATTRIBUTES_NEED_TANGENT
+        input.tangentOS = float4(vertexData.Tangent, 0);
+#endif
+    }
+}
+#endif
+
+//LBS Node not supported (skinning in vertex shader)
+
+#if defined(DOTS_INSTANCING_ON)
+// position only for motion vec vs
+void DOTS_Deformation_MotionVecPass(inout float3 currPos, inout float3 prevPos, uint vertexID)
+{
+    // x = curr frame index
+    // y = prev frame index
+    // z = deformation check (0 = no deformation, 1 = has deformation)
+    // w = skinned motion vectors
+    const int4 deformProperty = asint(unity_ComputeMeshIndex);
+    const int doSkinning = deformProperty.w;
+    if (doSkinning == 1)
+    {
+        const int currStreamIndex = _HybridDeformedVertexStreamIndex;
+		const int prevStreamIndex = (currStreamIndex + 1) % 2;
```

Comment:
I just wrote this quick an dirty. There is probably a more efficient way to flip this.